src/frontend/components/Header.vue

```diff
@@ -0,0 +1,229 @@
+import { useRouter } from 'vue-router'
+
+const router = useRouter()
+const isScrolled = ref(false)
+const isStoreReady = ref(false)
+const userMenuRef = ref<HTMLElement | null>(null)
+const showUserMenu = ref(false)
+const isAuthenticated = ref(false)
+const authStore = ref<ReturnType<typeof useAuthStore> | null>(null)
+const userName = ref('')
+const isMobileMenuOpen = ref(false)
+
+watch(
+  [isStoreReady, authStore],
+  ([ready, store]) => {
+    if (ready && store?.user) {
+      isAuthenticated.value = store.isAuthenticated
+      userName.value = store.user.username
+        || (store.user.email ? store.user.email.split('@')[0] : '')
+    }
+    else {
+      isAuthenticated.value = false
+      userName.value = ''
+    }
+  },
+  { immediate: true }
+)
+
+function updateScroll() {
+  isScrolled.value = window.scrollY > 20
+}
+
+function handleClickOutside(event: MouseEvent) {
+  const target = event.target as HTMLElement
+  if (userMenuRef.value && !userMenuRef.value.contains(target)) {
+    showUserMenu.value = false
+  }
+}
+
+function handleLogout() {
+  if (isStoreReady.value && authStore.value) {
+    authStore.value.logout()
+    isAuthenticated.value = false
+    userName.value = ''
+    router.push('/')
+    showUserMenu.value = false
+  }
+}
+
+onMounted(() => {
+  // Add click event listener for click-outside detection
+  document.addEventListener('click', handleClickOutside)
+
+  // Safe initialization of auth store
+  // Using setTimeout to ensure it runs after Pinia and plugins are initialized
+  setTimeout(() => {
+    try {
+      // Try to dynamically import the store
+      import('~/stores/auth')
+        .then((module) => {
+          const { useAuthStore } = module
+          authStore.value = useAuthStore()
+          isStoreReady.value = true
+        })
+        .catch((err) => {
+          console.error('Error importing auth store:', err)
+        })
+    }
+    catch (error) {
+      console.error('Could not access auth store in layout:', error)
+    }
+  }, 100)
+
+  // Initialize scroll event listener
+  window.addEventListener('scroll', updateScroll)
+  updateScroll() // Check initial scroll position
+})
+
+onUnmounted(() => {
+  // Remove event listeners
+  document.removeEventListener('click', handleClickOutside)
+  window.removeEventListener('scroll', updateScroll)
+
+  // Clear auth store reference
+  authStore.value = null
+  isStoreReady.value = false
+})
+</script>
+
+<template>
+  <nav
+    class="fixed top-0 left-0 right-0 z-50 transition-all duration-300"
+    :class="[isScrolled ? 'bg-white/80 backdrop-blur-lg shadow-sm' : '']"
+  >
+    <div class="max-w-7xl mx-auto px-4 sm:px-6 lg:px-8">
+      <div class="flex items-center justify-between h-16">
+        <!-- Logo -->
+        <NuxtLink to="/" class="flex items-center gap-2">
+          <img
+            src="/certo-logo-text.png"
+            alt="Certo logo"
+            class="h-10 w-auto"
+          />
```

Comment:
We should utilize the `<NuxtImg />` component here so we can streamline all of our images that we utilize in our app

You can see how to utilize it over here in the [HomeTrustees.vue](https://github.com/MarcoJul/certo/blob/63521596f96b0f3c24eac73b7c9008614053873b/src/frontend/components/home/HomeTrustees.vue#L11)